### PR TITLE
Eco 1147/main screen

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Create the new UI for the main screen
#### Technical description:
- Update according to new UI guidelines
- Update offer repository to support the new offer master list order.
- Removed unused res
#### Tests:
- Updated offer repo

### CODE REVIEW

File: core/src/main/java/com/kin/ecosystem/core/data/offer/OfferRepository.java [hunk 1]
```diff
@@ -91,12 +99,55 @@ public void onFailure(ApiException e) {
...
+			masterList.add(tutorial);
+		}
+
+		Iterator<Offer> nativeSpendItr = nativeSpendOfferList.getOffers().iterator();
+		Iterator<Offer> nativeEarnItr = nativeEarnOfferList.getOffers().iterator();
+		Iterator<Offer> earnMPItr = cachedOfferList.getOffers().iterator();
+
+		while (nativeSpendItr.hasNext() || nativeEarnItr.hasNext() || earnMPItr.hasNext()) {
+			// 1st nativeSpendItr, 2ns nativeEarnItr
+			if(nativeSpendItr.hasNext()) {
```
Comment (shayb) [inline core/src/main/java/com/kin/ecosystem/core/data/offer/OfferRepository.java:115]: isnt it clearer to read 2 separate loops for each type then one loop

File: core/src/main/java/com/kin/ecosystem/core/network/model/OfferList.java [hunk 2]
```diff
@@ -32,6 +32,14 @@ public boolean addAtIndex(final int index, @NonNull final Offer offer) {
         }
     }
 
+    public boolean add(@NonNull final Offer offer) {
+        if (offers != null) {
```
Comment (shayb) [inline core/src/main/java/com/kin/ecosystem/core/network/model/OfferList.java:36]: why check null if @Nonnull

File: sdk/src/main/java/com/kin/ecosystem/base/BaseViewHolder.java [hunk 3]
```diff
@@ -90,6 +97,14 @@ protected void setTextColor(@IdRes int viewId, @ColorInt int color) {
 		view.setTextColor(color);
 	}
 
+	protected void setVectorDrawable(@IdRes int viewId, @DrawableRes int vectorResID) {
+		if(!AppCompatDelegate.isCompatVectorFromResourcesEnabled()) {
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/base/BaseViewHolder.java:101]: what is the minimum android vector drawable works
is it ok with our minimum

File: sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt [hunk 4]
```diff
@@ -108,19 +93,19 @@ class EcosystemPresenter(private val authDataSource: AuthDataSource,
...
             when (visibleScreen) {
                 ScreenId.ORDER_HISTORY -> navigator?.navigateToOrderHistory(false)
-                ScreenId.MARKETPLACE, ScreenId.NONE -> navigator?.navigateToMarketplace(CustomAnimation.Builder().enter(R.anim.kinecosystem_slide_in_right).exit(R.anim.kinecosystem_slide_out_right).build())
-                else -> navigator?.navigateToMarketplace(CustomAnimation.Builder().enter(R.anim.kinecosystem_slide_in_right).exit(R.anim.kinecosystem_slide_out_right).build())
+                ScreenId.MARKETPLACE,
+                ScreenId.NONE ->
+                    navigator?.navigateToMarketplace()
+                else -> navigator?.navigateToMarketplace(CustomAnimation.Builder()
+                        .enter(R.anim.kinecosystem_slide_in_right)
+                        .exit(R.anim.kinecosystem_slide_out_right)
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt:101]: nice:)

File: sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.kt [hunk 5]
```diff
@@ -64,14 +58,14 @@ class EcosystemActivity : KinEcosystemBaseActivity(), IEcosystemView {
...
-                BlockchainSourceImpl.getInstance(), EventLoggerImpl.getInstance(), this, savedInstanceState, intent.extras).apply {
+                BlockchainSourceImpl.getInstance(), this, savedInstanceState, intent.extras).apply {
             onAttach(this@EcosystemActivity)
         }
     }
 
     @StyleRes
     private fun getKinTheme(): Int {
-        return when(ConfigurationImpl.getInstance().kinTheme!!) {
+        return when (ConfigurationImpl.getInstance().kinTheme!!) {
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.kt]: cant be set to null somewhere?

File: sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.kt [hunk 6]
```diff
@@ -0,0 +1,381 @@
+package com.kin.ecosystem.marketplace.presenter
+
+
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.kt:3]: whats the difference with the old file?

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.kt [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                BlockchainSourceImpl.getInstance(),
+                SettingsDataSourceImpl(SettingsDataSourceLocal(context)),
+                navigator,
+                EventLoggerImpl.getInstance())
+        marketplacePresenter?.onAttach(this)
+        return root
+    }
+
+    override fun onStart() {
+        super.onStart()
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.kt:52]: better i think to listen to on puase and resumed

File: sdk/src/main/java/com/kin/ecosystem/widget/TouchIndicatorIcon.java [hunk 8]
```diff
@@ -73,8 +73,7 @@ private void setIcon(final Drawable icon) {
 		icon.setBounds(0, 0, iconSize, iconSize);
 		drawables[0] = icon;
 		Drawable touchIndicator = ContextCompat.getDrawable(getContext(), R.drawable.kinecosystem_info_red_dot);
-		final int left = icon.getIntrinsicWidth() - indicatorRadius;
-		touchIndicator.setBounds(left, 0, indicatorRadius + left, indicatorRadius);
+		touchIndicator.setBounds(0, 0, indicatorRadius, indicatorRadius);
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/widget/TouchIndicatorIcon.java:76]: is this to make the click area bigger?

File: sdk/src/main/res/drawable/kinecosystem_ic_kin_logo_small_green.xml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    android:viewportHeight="12"
+    >
+
+    <group
+        android:name="kin_logo"
+        android:pivotX="6"
+        android:pivotY="6">
+
+        <path
+            android:pathData="M 8.571 4.679 L 9.261 4.679 C 9.389 4.681 9.516 4.648 9.628 4.585 C 9.74 4.522 9.833 4.431 9.898 4.321 C 9.964 4.21 9.999 4.084 10 3.956 L 10 2 L 7.965 2 C 7.837 1.998 7.71 2.03 7.598 2.093 C 7.486 2.156 7.393 2.247 7.327 2.357 C 7.262 2.468 7.226 2.594 7.225 2.722 L 7.225 3.365 C 7.225 4.091 6.662 4.715 5.92 4.715 C 5.178 4.715 4.54 4.091 4.54 3.365 L 4.54 2.697 C 4.539 2.573 4.505 2.452 4.442 2.346 C 4.378 2.239 4.288 2.151 4.181 2.09 C 4.073 2.03 3.951 1.999 3.827 2 L 2 2 L 2 10 L 3.813 10 C 4.004 10.002 4.187 9.928 4.323 9.795 C 4.46 9.662 4.538 9.48 4.54 9.289 L 4.54 8.687 C 4.54 7.923 5.116 7.281 5.858 7.281 L 5.858 7.283 C 6.6 7.283 7.226 7.896 7.226 8.621 L 7.226 9.283 C 7.226 9.679 7.554 10 7.959 10 L 10 10 L 10 8.022 C 9.999 7.895 9.964 7.77 9.899 7.661 C 9.834 7.551 9.741 7.461 9.631 7.398 C 9.52 7.336 9.394 7.304 9.267 7.305 L 8.657 7.305 C 7.915 7.305 7.226 6.717 7.226 5.992 L 7.228 5.992 C 7.228 5.267 7.829 4.679 8.571 4.679"
```
Comment (shayb) [inline sdk/src/main/res/drawable/kinecosystem_ic_kin_logo_small_green.xml:17]: const!

File: sdk/src/main/res/drawable/kinecosystem_ic_kin_logo_small_green.xml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+            android:fillColor="@color/kinecosystem_spend"
```
Comment (shayb) [inline sdk/src/main/res/drawable/kinecosystem_ic_kin_logo_small_green.xml:18]: just kidding

**Review by shayb — APPROVED** [review]

File: core/src/main/java/com/kin/ecosystem/core/data/offer/OfferRepository.java (hunk 1 above)
Comment (OrenZak) [reply core/src/main/java/com/kin/ecosystem/core/data/offer/OfferRepository.java:115]: No, the order is as follows: nativeSpendItr, nativeEarnItr, nativeSpendItr, nativeEarnItr.
and not all nativeSpendItr and then all nativeEarnItr.

File: core/src/main/java/com/kin/ecosystem/core/network/model/OfferList.java (hunk 2 above)
Comment (OrenZak) [reply core/src/main/java/com/kin/ecosystem/core/network/model/OfferList.java:36]: 👍

File: sdk/src/main/java/com/kin/ecosystem/base/BaseViewHolder.java (hunk 3 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/base/BaseViewHolder.java:101]: from what we tested 19+

File: sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.kt (hunk 5 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.kt]: will change to something else

File: sdk/src/main/java/com/kin/ecosystem/widget/TouchIndicatorIcon.java (hunk 8 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/widget/TouchIndicatorIcon.java:76]: it's to set it left top with the correct margins

File: sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.kt (hunk 6 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.kt:3]: removed  unused object and updated new logic related to new UI :)

**Pull request merged**